src/process_stash.cpp

```diff
@@ -0,0 +1,82 @@
+#include "process_stash.hpp"
+
+#include <boost/uuid/uuid.hpp>
+#include <boost/uuid/uuid_generators.hpp>
+#include <boost/uuid/uuid_io.hpp>
+
+#include <iterator>
+#include <shared_mutex>
+#include <unordered_map>
+
+namespace
+{
+    // A mapping containing handles to heterogenous objects.
+    std::unordered_map<std::string, boost::any> g_stash; // NOLINT(cppcoreguidelines-avoid-non-const-global-variables)
+
+    // A mutex which protects the map from data corruption.
+    std::shared_mutex g_mtx; // NOLINT(cppcoreguidelines-avoid-non-const-global-variables, cert-err58-cpp)
+
+    auto generate_unique_key() -> std::string
+    {
+        std::string uuid;
+        uuid.reserve(36); // NOLINT(cppcoreguidelines-avoid-magic-numbers, readability-magic-numbers)
+        uuid = to_string(boost::uuids::random_generator{}());
+
+        while (g_stash.find(uuid) != std::end(g_stash)) {
+            uuid = to_string(boost::uuids::random_generator{}());
+        }
+
+        return uuid;
+    } // generate_unique_key
+} // anonymous namespace
+
+namespace irods::http::process_stash
+{
+    auto insert(boost::any _value) -> std::string
+    {
+        std::lock_guard lock{g_mtx};
+        return g_stash.insert_or_assign(generate_unique_key(), std::move(_value)).first->first;
+    } // insert
+
+    auto find(const std::string& _key) -> boost::any
+    {
+        {
+            std::shared_lock lock{g_mtx};
+            if (auto iter = g_stash.find(_key); iter != std::end(g_stash)) {
+                return iter->second;
+            }
+        }
+
+        return nullptr;
```

Comment:
Based on the documentation, we have to check for `nullptr` now.

Is the use of `find` throughout the codebase safe? Do we need more checks?

Reply:
Don't think you should be able to access `nullptr` unless you do the following:

`boost::any_cast<std::nullptr_t>(null_bad);`

Could be wrong, might need more testing.
Either way, wanted to bring up since `nullptr` being there is kinda odd, even if it works.

Reply:
Yes, I agree. Seeing `nullptr` there definitely raised an eyebrow or two.

Reply:
Consider `std::optional<boost::any>`.

Reply:
Used `std::optional` approach.